### PR TITLE
Re-enable black_box_test framework on Android arm

### PR DESCRIPTION
**Pull request by maxz-lab**
Enable blackbox tests on Android arm

Some failed test cases are filtered and the list is added in configuration.py as global.
This effort is tracked by b/291648592.
Add "blackbox" test as a virtual test target.

b/165629644

### CODE REVIEW

File: starboard/build/platform_configuration.py [hunk 1]
```diff
@@ -177,3 +177,11 @@ def GetTestTargets(self):
         'slot_management_test',
         'starboard_platform_tests',
     ]
+
+  def GetTestVirtualTargets(self):
+    """Gets all tests to be run in a black box test run.
```
Comment (johnxwork) [inline starboard/build/platform_configuration.py:182]: Shall we unify the description of `GetTestVirtualTargets` across different overrides?

File: starboard/android/arm/cobalt/configuration.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+
+class CobaltAndroidArmConfiguration(configuration.CobaltAndroidConfiguration):
+  """Starboard Android Arm Cobalt configuration."""
+
+  def GetTestFilters(self):
+    filters = super().GetTestFilters()
+    for target, tests in _FILTERED_TESTS.items():
+      filters.extend(test_filter.TestFilter(target, test) for test in tests)
```
Comment (johnxwork) [inline starboard/android/arm/cobalt/configuration.py:52]: I bet the these tests fail on other Android architectures as well. But I guess we are only going to make sure tests pass on ARM first. So I have no problem with this.

**Review by johnxwork — CHANGES_REQUESTED** [review]

File: starboard/android/arm/cobalt/configuration.py (hunk 2 above)
Comment (maxz-lab) [reply starboard/android/arm/cobalt/configuration.py:52]: Yes, targeting ARM only for now.

File: starboard/build/platform_configuration.py (hunk 1 above)
Comment (maxz-lab) [reply starboard/build/platform_configuration.py:182]: Renamed to GetTestBlackBoxTargets and similar for other occasions.

File: starboard/build/application_configuration.py [hunk 3]
```diff
@@ -71,3 +71,11 @@ def GetTestTargets(self):
       A list of strings of test target names.
     """
     return []
+
+  def GetTestBlackBoxTargets(self):
+    """Gets all tests to be run in a unit test run.
```
Comment (johnxwork) [inline starboard/build/application_configuration.py:76]: Change "unit test" and "virtual test target" to match the new name?

**Review by johnxwork — APPROVED** [review]

**Pull request merged**